frontend/src/metabase/lib/time.js

```diff
@@ -40,9 +42,17 @@ const NUMERIC_UNIT_FORMATS = {
       .startOf("quarter"),
 };
 
+let hasSetMomentStartOfWeek = false;
+
 // only attempt to parse the timezone if we're sure we have one (either Z or ±hh:mm or +-hhmm)
 // moment normally interprets the DD in YYYY-MM-DD as an offset :-/
 export function parseTimestamp(value, unit = null, local = false) {
+  // make sure the moment start of week is set correctly the first time we parse something. updateMomentStartOfWeek()
+  // will return true if it was able to set it.
+  if (!hasSetMomentStartOfWeek) {
+    hasSetMomentStartOfWeek = updateMomentStartOfWeek();
+  }
+
```

Comment:
Why do we need to do this once here? I think the `MetabaseSettings.on` will always set the Moment setting if we have `start-of-week` set.

Reply:
The `on` listeners t only gets called when you call `Settings.set`. But we actually don't need this because `setLocalization` always gets called when the app loads which in turn calls `updateMomentStartOfWeek`